Use dbi's path to load dac and add RPATH to make files.

mikem8361 opened the pull request:
Fixes VS's problems with debugging and running different versions of coreclr.

mikem8361 commented:
@janvorli would you mind reviewing?  Thanks.

sergiy-k commented on src/ToolBox/SOS/Strike/CMakeLists.txt:
```diff
@@ -1,3 +1,11 @@
+# Set the RPATH of clrdbg so that it can find dependencies without needing to set LD_LIBRARY_PATH
```
Did you really mean to say "clrdbg" in this comment or this is just a copy-and-paste error?

sergiy-k commented on src/dlls/mscordbi/CMakeLists.txt:
```diff
@@ -1,3 +1,12 @@
+
+# Set the RPATH of clrdbg so that it can find dependencies without needing to set LD_LIBRARY_PATH
```
You have "clrdbg" in this comment too.

sergiy-k commented:
LGTM

janvorli commented:
LGTM modulo the clrdbg in the comments as Sergiy has pointed out.

mikem8361 commented:
@dotnet-bot test this please

mikem8361 commented:
@janvorli sorry, but could you re-review these changes. I've cleaned up PAL's module.cpp and fix a problem and simplified GetDACModule not to depend on the module instance handle.

janvorli commented on src/pal/src/loader/module.cpp:
```diff
@@ -1364,6 +1374,37 @@ static LPWSTR LOADGetModuleFileName(MODSTRUCT *module)
+    _ASSERTE(libraryNameOrPath != nullptr);
+    _ASSERTE(libraryNameOrPath[0] != '\0');
+
+    HMODULE dl_handle = dlopen(libraryNameOrPath, RTLD_LAZY);
```
A nit, it seems it would be cleaner to use void* instead of HMODULE here. The result is returned as void* and the dlopen returns void*.

janvorli commented:
LGTM modulo the nit.
